src/job_cache/eviction_policy.cpp

```diff
@@ -42,6 +42,62 @@
 
 namespace job_cache {
 
+struct TTLEvictionPolicyImpl {
+  std::string cache_dir;
+  job_cache::Transaction transact;
+  std::thread cleaning_thread;
+  job_cache::PreparedStatement jobs_older_than;
+  job_cache::PreparedStatement remove_jobs_older_than;
+
+  static constexpr const char* jobs_older_than_query =
+      "select job_id, cmd from jobs where create_time < ?";
+
+  static constexpr const char* remove_jobs_older_than_query =
+      "delete from jobs where create_time < ?";
+
+  TTLEvictionPolicyImpl(std::string dir, std::shared_ptr<job_cache::Database> db)
+      : cache_dir(dir),
+        transact(db),
+        jobs_older_than(db, jobs_older_than_query),
+        remove_jobs_older_than(db, remove_jobs_older_than_query) {}
+
+  void cleanup(int64_t seconds_to_live) {
+    std::vector<std::pair<int64_t, std::string>> jobs_to_remove;
+    // Compute the deadline, after which we collect everything
+    int64_t deadline = current_time_microseconds() - seconds_to_live * 1000000ll;
+
+    transact.run([this, &jobs_to_remove, deadline]() {
+      auto reset1 = wcl::make_defer([this]() { jobs_older_than.reset(); });
+      jobs_older_than.bind_integer(1, deadline);
+      // First find the use time that we want to remove from
+      while (jobs_older_than.step() == SQLITE_ROW) {
+        int64_t job_id = jobs_older_than.read_integer(0);
+        std::string cmd = jobs_older_than.read_string(1);
+        for (auto& ch : cmd) {
+          if (ch == '\0') ch = ' ';
+        }
+        jobs_to_remove.emplace_back(job_id, std::move(cmd));
```

Comment:
why do we care about the command here? Isn't the job_id sufficient?

Reply:
We want to log the command of the job that we're evicting